libs/vm/include/vm/module.hpp

```diff
@@ -231,6 +205,43 @@ class Module
     }
 
   private:
+    template <typename GetterReturnType, typename... GetterArgs, typename SetterReturnType,
+              typename... SetterArgs>
+    ClassInterface &InternalEnableIndexOperator(GetterReturnType (Type::*getter)(GetterArgs...),
+                                                SetterReturnType (Type::*setter)(SetterArgs...))
+    {
+      static_assert(sizeof...(GetterArgs) >= 1, "Getter should take at least one parameter");
+      static_assert(sizeof...(SetterArgs) >= 2, "Setter should take at least two parameters");
+      static_assert(
+          std::is_same<GetterReturnType,
+                       std::remove_cv_t<std::remove_reference_t<
```

Comment:
does `std::decay` make sense here?

Reply:
Just to record our conversation just now: I don't think it does.